warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RangeConjunctionRebuildingVisitor.java

```diff
@@ -92,6 +92,9 @@ public RangeConjunctionRebuildingVisitor(ShardQueryConfiguration config, Scanner
                 throw new DatawaveFatalQueryException(qe);
             }
             
+            // inject reference nodes when needed
+            script = (T) EnsureReferenceNodesVisitor.ensureReferences(script);
```

Comment:
Is some visitor breaking this contract such that we need to do this?

Reply:
The RangeConjunctionRebuildingVisitor can produce a reference expression without a reference node